lib/matplotlib/category.py

```diff
@@ -148,11 +148,11 @@ def __init__(self, units_mapping):
         self._units = units_mapping
 
     def __call__(self, x, pos=None):
-        if pos is None:
-            return ""
-        r_mapping = {v: StrCategoryFormatter._text(k)
-                     for k, v in self._units.items()}
-        return r_mapping.get(int(np.round(x)), '')
+        return '' if pos is None else self.format_ticks([x])[0]
+
+    def format_ticks(self, values):
```

Comment:
I'm curious/confused about why/how the dict is only built once if there's still a call to .format_ticks([x]) on every __call__ to call. Is their caching somewhere?

Reply:
the tick-labeling code calls `format_ticks`, not `__call__` (`__call__` is only used by the mouse-cursor-text (as a fallback because `format_data_short` is not defined) and left as a backcompat API).

Reply:
:/ I have no objection to switching out the data structure  holding the mapping to something more reverse friendly

Reply:
the deprecation is a bit a pain to handle, I doubt it's worth it, especially considering that nearly all cases where this would be relevant are actually due to user error (not converting strings to floats/dates).